scripts/update_collection_db.py

```diff
@@ -0,0 +1,108 @@
+"""
```

Comment:
It may be preferable to refactor this file so that consistency verification and repair happens in one direction at a time:
  - check that s3 is consistent with ddb
  - check that ddb is consistent with s3

Each operation, separately, is parallelizable.

Reply:
So, current behavior consists of these separate steps (though these 3 are all in `def __init__`):
1. Look at all collections in s3 and make a set of uuids.
2. Look at all table items in dynamodb and make a set of uuids.
3. Make a set of uuids in s3 that aren't in the dynamodb table.

Then you can either hard reset:
4a. Delete all items in the DB and rebuild all collections with everything from the bucket.

Or update:
4b. Delete nothing, add all collection entries from s3 that aren't already in the database.
Note: GET /collections operations check the file before returning and delete the entry from the database if it doesn't exist.  If an entry is somehow missed, it won't break anything and should eventually be removed through normal use if it's not in the bucket.

If you want to guarantee the entries are 1-to-1, the hard reset seems fairly efficient to me (it's always going to be expensive; every file will need to be opened once to check ownership no matter what you do).  Otherwise, updating the table with only the diff can be much faster and should never break anything.

We should talk on Monday!